src/main/java/appeng/fluids/client/gui/GuiFluidTerminal.java

```diff
@@ -229,22 +228,23 @@ protected void handleMouseClick( Slot slot, int slotIdx, int mouseButton, ClickT
 	{
 		if( slot instanceof SlotFluidME )
 		{
-			SlotFluidME meSlot = (SlotFluidME) slot;
+			final SlotFluidME meSlot = (SlotFluidME) slot;
+
 			if( clickType == ClickType.PICKUP )
 			{
 				// TODO: Allow more options
 				if( mouseButton == 0 && meSlot.getHasStack() )
 				{
-					this.container.setTargetStack( meSlot.getAEStack() );
-					AELog.info( "mouse0 GUI STACK SIZE %s", meSlot.getAEStack().getStackSize() );
+					this.container.setTargetStack( meSlot.getAEFluidStack() );
+					AELog.info( "mouse0 GUI STACK SIZE %s", meSlot.getAEFluidStack().getStackSize() );
 					NetworkHandler.instance().sendToServer( new PacketInventoryAction( InventoryAction.FILL_ITEM, slot.slotNumber, 0 ) );
 				}
 				else
 				{
-					this.container.setTargetStack( meSlot.getAEStack() );
-					if( meSlot.getAEStack() != null )
+					this.container.setTargetStack( meSlot.getAEFluidStack() );
+					if( meSlot.getAEFluidStack() != null )

```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS134)